pkgs/development/compilers/llvm/17/mlir/default.nix

```
@@ -0,0 +1,65 @@
+{ lib, stdenv, llvm_meta
+, buildLlvmTools
+, monorepoSrc, runCommand
+, cmake
+, ninja
+, libxml2
+, libllvm
+, version
+, doCheck ? !stdenv.hostPlatform.isMusl
+}:
+
+stdenv.mkDerivation rec {
+  pname = "mlir";
+  inherit version doCheck;
+
+  # Blank llvm dir just so relative path works
+  src = runCommand "${pname}-src-${version}" {} ''
+    mkdir -p "$out"
+    cp -r ${monorepoSrc}/cmake "$out"
+    cp -r ${monorepoSrc}/${pname} "$out"
```

```suggestion
    cp -r ${monorepoSrc}/mlir "$out"
```

This doesn't depend on `pname`. Imagine if `pname` is replaced by `mlir-unwrapped`.